diff -Nurd unburden-home-dir-0.4.0.1.orig/bin/unburden-home-dir unburden-home-dir-0.4.0.1/bin/unburden-home-dir
--- a/bin/unburden-home-dir	2016-05-25 09:57:15.000000000 +1000
+++ b/bin/unburden-home-dir	2016-12-18 14:40:56.113915063 +1100
@@ -28,18 +28,16 @@

 # Globally define version
 our $VERSION = '0.4.0.1';
+my $VARNAME_MATCH = qr/\$([A-Z_][A-Z0-9_]*|\{.*?\})/;

 # Load Modules
-use Config::File;
 use Getopt::Std; $Getopt::Std::STANDARD_HELP_VERSION = 1;
 use File::Path qw(mkpath rmtree);
 use File::Basename;
 use File::BaseDir qw(config_home);
 use File::Touch;
-use File::Rsync;
 use File::Which;
 use IO::Handle;
-use String::Expand;
 use Data::Dumper;

 # Determine default value for target directory
@@ -140,6 +138,112 @@
     return;
 }

+#
+# BEGIN: Non-packaged lighter-weight implementations of read_config_file and rsync
+#
+sub expand_one_var($$) {
+  my ($var, $vars) = @_;
+
+  $var =~ s/^\{(.*)\}$/$1/;
+  croak("Unknown variable '$var'") unless defined $vars->{$var};
+
+  return $vars->{$var};
+}
+
+sub expand_strings_one_var($$$$) {
+  my ( $var, $strs, $overlay, $done ) = @_;
+
+  # Chop off delimiting {braces} if present
+  $var =~ s/^\{(.*)\}$/$1/;
+
+  if( exists $strs->{$var} ) {
+    return $strs->{$var} if $done->{$var};
+    croak("Variable loop trying to expand '$var'") if exists $done->{$var};
+    $done->{$var} = 0;
+    expand_strings_inner($strs, $overlay, $var, $done);
+    return $strs->{$var};
+  }
+
+  return $overlay->{$var} if exists $overlay->{$var};
+
+  croak("Unknown variable '$var'");
+}
+
+sub expand_strings_inner($$$$) {
+  my ( $strs, $overlay, $v, $done ) = @_;
+
+  if($strs->{$v} =~ m/[\\\$]/) {
+    $strs->{$v} =~ s{\\([\\\$])|$VARNAME_MATCH}{$1 or expand_strings_one_var($2, $strs, $overlay, $done)}eg;
+  }
+
+  $done->{$v} = 1;
+}
+
+sub expand_string($$) {
+  my ($str, $vars) = @_;
+
+  $str =~ s{\\([\\\$])|$VARNAME_MATCH}{$1 or expand_one_var($2, $vars)}eg;
+
+  return $str;
+}
+
+sub expand_strings($$) {
+  my ($strs, $overlay) = @_;
+
+  my %done;
+
+  foreach my $v (keys %$strs) {
+    expand_strings_inner( $strs, $overlay, $v, \%done );
+  }
+}
+
+# Read configuration file
+sub read_config_file($) {
+  my $file = shift;
+  my $config = {};
+  my $line_num = 0;
+
+  open(FH, $file) or croak("Can't read configuration in $file: $!\n");
+
+  while (++$line_num and my $line = <FH>) {
+    # remove new lines
+    chomp($line);
+
+    # skip comments and blank lines
+    next if $line =~ m/^(\s+|\s*#.*)$/;
+
+    unless ($line =~ m/\s*([A-Za-z0-9_\-]+)\s*=\s*(['"]?)(.*?)\2\s*$/) {
+      warn "Invalid format at line $line_num, skipping.";
+      next;
+    }
+
+    # assign the key/value
+    $config->{$1} = $3;
+  }
+
+  close(FH);
+
+  return $config;
+}
+
+# Perform an verbose archive via rsync
+sub rsync($$) {
+  my ($from, $to) = @_;
+
+  open(PS, "rsync -va \"$from\" \"$to\" 2>&1 |") || return 1;
+
+  while (<PS>) {
+    chomp;
+    say unless m(^sent |^total size|^\s*$)
+  }
+
+  return $? >> 8;
+}
+
+# END
+#
+
+
 # Parse command line options
 getopts('hnuf:Fb:c:C:l:L:', \%OPTIONS);
 
@@ -182,7 +286,7 @@
 foreach my $configfile (@CONFFILES) {
     if ( -e $configfile ) {
         # Workaround RT#98542 in Config::File 1.50 and earlier
-        my $cf = Config::File::read_config_file($configfile);
+        my $cf = read_config_file($configfile);
         if (defined($cf)) {
             $CONFIG = { %$CONFIG, %$cf };
         }
@@ -206,22 +310,6 @@
 my $OLDUMASK = umask();
 umask(077);
 
-# Initialize rsync object
-my $rsync = File::Rsync->new(
-    archive => 1,
-    verbose => 1,
-    outfun => sub {
-        my $output = shift;
-        chomp($output);
-        say $output unless $output =~ m(^sent |^total size|^\s*$);
-    },
-    errfun => sub {
-        # uncoverable subroutine
-        chomp;          # uncoverable statement
-        warn "$_[0]\n"; # uncoverable statement
-    },
-);
-
 # Check for lsof in search path
 my $which_lsof = which('lsof');
 # Extra check for crappy distributions which place lsof outside a
@@ -257,10 +345,7 @@
             $from .= '/';
             $to .= '/';
 
-            my $rc = $rsync->exec(
-                src => $from,
-                dst => $to,
-            );
+            my $rc = rsync($from, $to);
             rmtree($from);
         } else {
             my $rc = system(qw(mv -v), $from, $to);
